test/e2e/storage/in_tree_volumes.go

```diff
@@ -0,0 +1,455 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/rand"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+)
+
+var testDrivers = map[string]func() testDriver{
+	"NFS":                        initNFSDriver,
+	"GlusterFS":                  initGlusterFSDriver,
+	"iSCSI [Feature:Volumes]":    initISCSIDriver,
+	"Ceph-RBD [Feature:Volumes]": initRbdDriver,
+	"CephFS [Feature:Volumes]":   initCephFSDriver,
+	"hostpath":                   initHostpathDriver,
+	"hostpathSymlink":            initHostpathSymlinkDriver,
+	"emptydir":                   initEmptydirDriver,
+	"cinder":                     initCinderDriver,
+	"gce":                        initGceDriver,
+	"vSphere":                    initVSphereDriver,
+	"azure":                      initAzureDriver,
+	"aws":                        initAwsDriver,
+}
+
+var testSuites = map[string]func() testSuite{
```

Comment:
Can this just be a slice without the name of the test suite?  It would be better if the test suite name was defined in the test suite itself (via Context()) instead of in the caller

Reply:
Changed as suggested.